charm-slurmd/src/charm.py

```diff
@@ -105,6 +105,25 @@ def __init__(self, *args):
 
     def _on_install(self, event):
         """Perform installation operations for slurmd."""
+        logger.debug(f"### Relocating NHC Resource to local directory")
+        try:
+            nhc_path = self.model.resources.fetch("nhc")
+            logger.debug(f"## Found nhc resource: {nhc_path}")
+        except Exception as e:
+            logger.error("## Missing nhc resource")
+            self.unit.status = BlockedStatus("Missing nhc resource")
+            event.defer()
+            return
+        # move the nhc tarball to temp directory for later acess by slurm ops manager
+        nhc_tmp_file = "/opt/nhc.tar.gz"
+        content = ""
+        with open(nhc_path, "rb") as f:
+            content = f.read()
+        f = open(nhc_tmp_file, 'wb')
+        f.write(content)
+        f.close()
+        logger.debug(f"### Relocating NHC Resource Successful")
```

Comment:
```suggestion
```

This is not needed, the variable `nhc_path` will contain a valid path for the resource.

I suggest you pass it to the `successful_installation = self._slurm_manager.install(custom_repo, nhc_path)` line below and extract it in the slurm-ops-manager code.